File: python/easy/1470_Shuffle_the_Array.py
from typing import *
class Solution:
    # 60 ms, faster than 61.39% of Python3 online submissions for Shuffle the Array.
    # 14.2 MB, less than 98.76% of Python3 online submissions for Shuffle the Array.
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        ans = []
        for i in range(n):
            ans.append(nums[i])
            ans.append(nums[n + i])
        return ans

    # 56 ms, faster than 84.56% of Python3 online submissions for Shuffle the Array.
    # 14.2 MB, less than 98.76% of Python3 online submissions for Shuffle the Array.
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        ans = []
        for x, y in zip(nums[:n], nums[n:]):
            ans.append(x)
            ans.append(y)
        return ans

    #  60 ms, faster than 61.39% of Python3 online submissions for Shuffle the Array.
    # 14.5 MB, less than 49.20% of Python3 online submissions for Shuffle the Array.
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        left = nums[:n]
        right = nums[n:]

        ans = [ None ]* (n *2)
        ans[::2] = left
        ans[1::2] = right
        return ans

    # 60 ms, faster than 61.39% of Python3 online submissions for Shuffle the Array.
    # 14.5 MB, less than 49.20% of Python3 online submissions for Shuffle the Array.
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        ans = [ None ] * ( n* 2)
        ans[::2] = nums[:n]
        ans[1::2] = nums[n:]
        return ans


if __name__ == "__main__":
    so = Solution()
    print(so.shuffle(nums = [2,5,1,3,4,7], n = 3))

File: c/easy/1470_Shuffle_the_Array.c
#include <stdio.h>
#include <stdlib.h>

int* shuffle(int* nums, int numsSize, int n, int* returnSize);

int main(void){
    int nums[] = {2,5,1,3,4,7};
    int n = 3;
    int returnSize = 0;

    int * shuffled_nums = shuffle(nums, n, n, &returnSize);

    for (int i =0; i < returnSize; i++){
        printf(" %3d ", shuffled_nums[i]);
    }
    printf("\n");

    free(shuffled_nums);

    return 0;
}

// 28 ms, faster than 7.93% of C online submissions for Shuffle the Array.
// 7.1 MB, less than 52.24% of C online submissions for Shuffle the Array.
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    int * shuffled_nums = (int *) malloc( sizeof(int) * n * 2);
    int i;
    int j=0;

    *returnSize = n * 2;
    for (i = 0; i < n; i++){
        shuffled_nums[j] = nums[i];
        shuffled_nums[j+1] = nums[i+n];
        j+=2;
    }
    return shuffled_nums;
}